default/python/AI/ProductionAI.py

```diff
@@ -1611,59 +1272,57 @@ def already_has_completed_colony_building(planet_id) -> bool:
     return any(universe.getBuilding(bldg).name.startswith("BLD_COL_") for bldg in planet.buildingIDs)
 
 
-def build_ship_facilities(bld_name, top_locs=None):
+def _build_ship_facilities(building_type: BuildingType, top_pids: Set[PlanetId] = None) -> None:
```

Comment:
You could use immutable default here.

```suggestion
def _build_ship_facilities(building_type: BuildingType, top_pids: Set[PlanetId] = frozenset()) -> None:
 ```

Reply:
good idea, safes the None-handling